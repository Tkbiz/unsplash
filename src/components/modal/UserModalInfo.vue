<template>
  <v-col cols="12"
         class="py-0 mb-9">
    <v-row
        v-if="userCard"
        align="center">
      <v-avatar
          v-if="userCard.profile_image"
          @click.prevent="openUserProfile(userCard.id)"
          size="84">
        <img
            :src="userCard.profile_image.large ? userCard.profile_image.large : '@/assets/images/no-image.png'"
            alt="John">
      </v-avatar>
      <div>
        <div v-if="userCard.name"
             @click.prevent="openUserProfile(userCard.id)"
             class="avatar-title">
          {{ userCard.name }}
        </div>
        <div v-if="userCard.location"
             class="avatar-nick">
          {{ userCard.location }}
        </div>
      </div>
    </v-row>
  </v-col>
</template>

<script>

export default {
  name: 'UserModalInfo',
  props: {
    userCard: {
      type: Object,
      required: true
    }
  },
  methods: {
    async openUserProfile(id) {
      if (this.userCard.username) {
        await this.$store.dispatch('fetchUserInfo', this.userCard.username)
        await this.$router.push(`/user-account/${id}`)
      }
    }
  }
}
</script>
