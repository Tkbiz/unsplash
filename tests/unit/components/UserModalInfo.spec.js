import {mount, createLocalVue} from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import UserModalInfo from '@/components/modal/UserModalInfo'

Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuex)

describe('UserModalInfo.vue', () => {
  let vuetify
  let wrapper
  let actions
  let store

  const $route = {
    path: '/user-account/1'
  }
  const mockPush = jest.fn()
  const $router = {
    push: mockPush
  }

  beforeEach(() => {
    vuetify = new Vuetify()

    actions = {
      fetchUserInfo: jest.fn()
    }
    store = new Vuex.Store({
      actions
    })

    wrapper = mount(UserModalInfo, {
      localVue,
      vuetify,
      propsData: {
        userCard: {
          username: 'username',
          profile_image: {},
          name: 'name'
        }
      },
      store,
      mocks: {
        $route,
        $router
      }
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('renders props when passed', () => {
    const userCard = {
      username: 'username',
      profile_image: {},
      name: 'name'
    }

    expect(wrapper.props()).toEqual({userCard})
  })

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("dispatches a fetchUserInfo action when avatar image is clicked", async () => {
    const username = 'username'

    wrapper.find('.v-avatar').trigger('click')
    await wrapper.vm.$nextTick()

    expect(actions.fetchUserInfo.mock.calls).toHaveLength(1)
    expect(actions.fetchUserInfo.mock.calls[0][1]).toEqual(username)
    expect(actions.fetchUserInfo).toHaveBeenCalled()
  })

  it("should go to user-account page when avatar image is clicked", async () => {
    wrapper.find('.v-avatar').trigger('click')
    await wrapper.vm.$nextTick()

    expect(mockPush).toBeCalled()
  })

  it("dispatches a fetchUserInfo action when avatar image is clicked", async () => {
    const username = 'username'

    wrapper.find('.avatar-title').trigger('click')
    await wrapper.vm.$nextTick()

    expect(actions.fetchUserInfo.mock.calls).toHaveLength(1)
    expect(actions.fetchUserInfo.mock.calls[0][1]).toEqual(username)
    expect(actions.fetchUserInfo).toHaveBeenCalled()
  })

  it("should go to user-account page when avatar title is clicked", async () => {
    wrapper.find('.avatar-title').trigger('click')
    await wrapper.vm.$nextTick()

    expect(mockPush).toBeCalled()
  })
})
