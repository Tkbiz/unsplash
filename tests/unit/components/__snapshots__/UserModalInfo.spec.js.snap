// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModalInfo.vue should match snapshot 1`] = `
<div class="py-0 mb-9 col col-12">
  <div class="row align-center">
    <div class="v-avatar" style="height: 84px; min-width: 84px; width: 84px;"><img src="@/assets/images/no-image.png" alt="John"></div>
    <div>
      <div class="avatar-title">
        name
      </div>
      <!---->
    </div>
  </div>
</div>
`;
